src/viam/examples/modules/complex/gizmo/api.hpp

```diff
@@ -44,6 +43,13 @@ class Gizmo : public Component {
     explicit Gizmo(std::string name);
 };
 
+template <>
+struct API::api_map<Gizmo> {
```

Comment:
Cool idea! Some qs: what's up with the `template <>`? Does this pattern ensure at compilation time that all derived classes of `Resource` implement a static `api` method?

Reply:
Ahh yeah, good question. I think there is nothing that necessarily enforces that we get a compilation error. We'll get a runtime error if we try to call `API::for_t<lightsaber>()` but it will compile. I added the additional `test_get_api` tests for each resource to create in part to create another point of contact where someone implementing a new resource would realize that adding this template is necessary.

Another option would be to do this template creation within the `ResourceRegistration` class, which would slot nicely in with some refactors of the (already templated) `ResourceRegistration2` class. But this seems a bit more complicated and worthy of pulling out into a separate PR imo.

Reply:
> Another option would be to do this template creation within the ResourceRegistration class, which would slot nicely in with some refactors of the (already templated) ResourceRegistration2 class.

Yeah this sounds like a valuable add for [RSDK-3030](https://viam.atlassian.net/browse/RSDK-3030) once we revisit that class. Would you mind leaving a `TODO` in `ResourceRegistration2` to think about an even further templated static `API` method? Otherwise, this PR seems like a solid step in the right direction, thank you! Curious what @acmorrow thinks.

Reply:
I think you would get a compile error on `API::for_t<Lightsaber>` if there was no specialization, not a runtime error, as long as there isn't non-specialized version that would fall back to some runtime behavior. I guess I'll see when I get further into the review.

Reply:
Ahh yes, you're absolutely right. Sorry, I had two ideas mixed up in my head! The concern I have is, if we don't implement the template specialization for `Foo` and _also_ don't call `API::for_t<Foo>` anywhere in the SDK, then `API::for_t<Foo>` just won't work but the I believe the code will compile, which I could see being a problem for users/module developers.

Reply:
I think that was true before though: if you failed to implement `static_api` for some type, but also never called it (or 
called anything that called it), everything would still compile but it wouldn't be useable for clients.